android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/BaseSearchTelemetry.kt

```diff
@@ -21,20 +25,47 @@ import mozilla.components.support.base.facts.collect
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.kotlinx.coroutines.flow.filterChanged
 import org.json.JSONObject
+import java.io.File
+
+internal const val COLLECTION_NAME = "search-telemetry-v2"
 
 /**
  * Main configuration and functionality for tracking ads / web searches with specific providers.
  */
-abstract class BaseSearchTelemetry {
+abstract class BaseSearchTelemetry(private val file: File) {
+    @VisibleForTesting
+    internal var providerList: List<SearchProviderModel>? = emptyList()
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    internal fun initializeProviderList(): Job {
+        val fetcher = SerpTelemetryFetcher(file, COLLECTION_NAME)
+
+        return CoroutineScope(Dispatchers.IO).launch {
+            providerList = fetcher.fetchSearchProviders()
+        }
+    }
```

Comment:
I think we could call this function directly from the `install` function to add some clarity around the order of operations. I don't think we need to return the `Job` either, and can instead treat this as a "fire and forget" function.

To summarize this with all of my comments below, my suggestion would be something like this:

```
// in this class
internal var providerList: List<SearchProviderModel>? = emptyList()

abstract fun install(engine: Engine, store: BrowserStore)

internal suspend fun initializeProviderList() {
    providerList = SerpTelemetryFetcher(file, COLLECTION_NAME).fetchSearchProviders()
}

// in Core.kt

            CoroutineScope(Dispatchers.Default).launch {
                // Install the "ads" WebExtension to get the links in an partner page.
                adsTelemetry.install(engine, this)

                // Install the "cookies" WebExtension and tracks user interaction with SERPs.
                searchTelemetry.install(engine, this)
            }

// in AdsTelemetry and InContentTelemetry
        val info = ExtensionInfo(
              ...
        )
        initializeProviderList()
        installWebExtension(engine, store, info)
```